dot/core/messages.go

```diff
@@ -81,6 +81,10 @@ func (s *Service) HandleTransactionMessage(peerID peer.ID, msg *network.Transact
 	isValid := true
 	for _, tx := range txs {
 		validity, isValidTxn, err := s.validateTransaction(peerID, head, rt, tx)
+		if err != nil {
+			return false, err
```

Comment:
feel free to add more context to the wrapping

```suggestion
			return false, fmt.Errorf("failed validating transaction for peerID %s: %w:", peerID, err)
```